src/common/json.hpp

```diff
@@ -124,19 +124,43 @@ class Json {
 	bool IsObject() const;
 	bool IsArray() const;
 	bool IsString() const;
-	bool IsInt() const;
+	bool IsInt64() const;
 	bool IsNumber() const;
 	bool IsDouble() const;
 	bool IsBool() const;
 	bool IsNull() const;
 
 	ExpectedString GetString() const;
-	ExpectedInt64 GetInt() const;
+	ExpectedInt64 GetInt64() const;
 	ExpectedDouble GetDouble() const;
 	ExpectedBool GetBool() const;
 
+	// Defined in cpp file as specialized templates.
 	template <typename T>
-	expected::expected<T, error::Error> Get() const;
+	typename enable_if<
+		not is_integral<T>::value or is_same<T, int64_t>::value,
```

Comment:
Hehe, yes. But to be fair I never said we won't use them, I said we won't *make* them. But that's exactly what I'm doing here, so I guess I'm still guilty. 😄

But laughing aside, I believe it has really improved the code, especially in this module. It's much less code when the type is also a parameter and can be passed around. It has come a long way since what I knew from before this project (which was pre-C++11).